pkg/util/yamltool/yamltool.go

```diff
@@ -0,0 +1,23 @@
+package yamltool
+
+import (
+	"fmt"
+
+	"gopkg.in/yaml.v2"
+)
+
+func Marshal(in interface{}) []byte {
+	buff, err := yaml.Marshal(in)
+	if err != nil {
+		panic(fmt.Errorf("marshal %s to yaml string failed: %v", in, err))
+	}
+	return buff
+}
+
+func Unmarshal(in []byte, out interface{}) {
```

Comment:
Proposal to return an error instead of using `panic`, I don't think using `panic` in a `util` usage package is a good idea.  The `util` package will be widely used in the project and a toolkit purpose function calls should not cause an `application immediately crash` consequence when the caller forgets to use `recover`[1]. 

## Reference
1. https://stackoverflow.com/questions/44504354/should-i-use-panic-or-return-error/44505268

Reply:
Here's the reason:

This utility aims to fix so many time calling `marshal && if err != nil` and `unmarshal && err !=  nil` in httppipeline/spec.go, httppipeline/httppipeline.go and supervisor/spec.go.  It saves a lot of repeated code.

And I've added comments to them, can you accept it after that?

Reply:
* I got it. Then propose to add the panic behavior comment for `yamltool. Marshal`, `yamltool. Unmarshal` as well. To make sure the caller is able to read this designed behavior through IDE's remind.
* I believe it's the caller's duty to read the toolkit functions' usage comment carefully before using it.

Reply:
Hmm, I had done it.

Reply:
Hmmm, can't see the `Outdated` tag in this GitHub conversation card. (So I thought u haven't updated this file)
My bad.